Deprovisioning Notification (#6672)
owncloud/ocis: services/userlog/README.md

Additionally to the oc10 API, the `userlog` service also provides an `/sse` (Server-Sent Events) endpoint to be informed by the server when an event happens. See [What is Server-Sent Events](https://medium.com/yemeksepeti-teknoloji/what-is-server-sent-events-sse-and-how-to-implement-it-904938bffd73) for a simple introduction and examples of server sent events. The `sse` endpoint will respect language changes of the user without needing to reconnect. Note that SSE has a limitation of six open connections per browser which can be reached if one has opened various tabs of the Web UI pointing to the same Infinite Scale instance.

## Posting

> The userlog service is able to store global messages that will be displayed in the WebUI to all users. These messages can only be activated and deleted by users with the `admin` role but not by ordinary users. If a user deletes the message in the WebUI, it reappears on reload. Global messages use the endpoint: `/ocs/v2.php/apps/notifications/api/v1/notifications/global` and are activated by sending a `POST` request. Note that Sending another `POST` request of the same type overwrites the previous one. For the time being, only the type `deprovision` is supported.

Reviewer: ```suggestion
The userlog service is able to store global messages that will be displayed in the Web UI to all users. These messages can only be activated and deleted by users with the `admin` role but not by ordinary users. If a user deletes the message in the Web UI, it reappears on reload. Global messages use the endpoint `/ocs/v2.php/apps/notifications/api/v1/notifications/global` and are activated by sending a `POST` request. Note that sending another `POST` request of the same type overwrites the previous one. For the time being, only the type `deprovision` is supported.
```

> ## Posting
> 
> The userlog service is able to store global messages that will be displayed to all users and cannot be deleted by them. These messages can be activated by sending a `POST` request to `/ocs/v2.php/apps/notifications/api/v1/notifications/global`. Only one type is supported as of now: `"deprovision"`. Only an user with the admin role is able to call this endpoint.
> 
> ### Deprovision
> 
> Deprovision messages announce the deprovision date of the platform to all users. The text of the message can be changed via translations. (See `Translations` section.) The endpoint expects only a `deprovision_date` parameter in the `POST` request body. It expects the string being in `RFC3339` format. However this format can be changed by using the `deprovision_date_format`. See [go time formating](https://pkg.go.dev/time#pkg-constants) for details.

Reviewer 2: ```suggestion
## Posting

The userlog service is able to store global messages that will be displayed in the WebUI to all users. These messages can only be activated and deleted by users with the admin role but not by ordinary users. If a user deletes the message in the WebUI, it reappears on reload. Global messages use the endpoint: `/ocs/v2.php/apps/notifications/api/v1/notifications/global` and are activated by sending a `POST` request. For the time being, only the type `deprovision` is supported.

### Deprovisioning

Deprovision messages announce a deprovision text including a deprovision date of the instance to all users. With this message, users get informed that the instance will be shut down and deprovisioned and no further access to their data is possible past the given date. This implies that users must download their data before the given date, the text shown to users refers to this information. Note that the task to deprovision the instance is not dependent on the message. The text of the message can be translated according the translation settings, see section [Translations](#translations). The endpoint only expects a `deprovision_date` parameter in the `POST` request body as the final text is assembled automatically. The string hast to be in `RFC3339` format, however, this format can be changed by using `deprovision_date_format`. See [go time formating](https://pkg.go.dev/time#pkg-constants) for details.
```

## Deleting

To delete events for an user, use a `DELETE` request to `ocs/v2.php/apps/notifications/api/v1/notifications` containing the IDs to delete.

> Additionally an user with the `admin` role call can send a `DELETE` request to `ocs/v2.php/apps/notifications/api/v1/notifications/global` to remove a global message. (See `Posting` section)

Reviewer 2: ```suggestion
Only users with the `admin` role can send a `DELETE` request to the `ocs/v2.php/apps/notifications/api/v1/notifications/global` endpoint to remove a global message, see the [Posting](#posting) section for more details.)
```

> Deprovision messages announce a deprovision text including a deprovision date of the instance to all users. With this message, users get informed that the instance will be shut down and deprovisioned and no further access to their data is possible past the given date. This implies that users must download their data before the given date, the text shown to users refers to this information. Note that the task to deprovision the instance is not dependent on the message. The text of the message can be translated according the translation settings, see section [Translations](#translations). The endpoint only expects a `deprovision_date` parameter in the `POST` request body as the final text is assembled automatically. The string hast to be in `RFC3339` format, however, this format can be changed by using `deprovision_date_format`. See the [go time formating](https://pkg.go.dev/time#pkg-constants) for more details.

Reviewer: ```suggestion
Deprovision messages announce a deprovision text including a deprovision date of the instance to all users. With this message, users get informed that the instance will be shut down and deprovisioned and no further access to their data is possible past the given date. This implies that users must download their data before the given date. The text shown to users refers to this information. Note that the task to deprovision the instance does not depend on the message. The text of the message can be translated according to the translation settings, see section [Translations](#translations). The endpoint only expects a `deprovision_date` parameter in the `POST` request body as the final text is assembled automatically. The string hast to be in `RFC3339` format, however, this format can be changed by using `deprovision_date_format`. See the [go time formating](https://pkg.go.dev/time#pkg-constants) for more details.
```
